include/eve/detail/function/simd/arm/sve/load.hpp

```diff
@@ -15,28 +15,83 @@
 
 namespace eve::detail
 {
-template<real_scalar_value T, typename N, simd_compatible_ptr<wide<T, N>> Ptr>
+template<relative_conditional_expr C,
+         real_scalar_value         T,
+         typename N,
+         simd_compatible_ptr<wide<T, N>> Ptr>
 EVE_FORCEINLINE wide<T, N>
-load_(EVE_SUPPORTS(sve_), ignore_none_ const&, safe_type const&, eve::as<wide<T, N>> const&, Ptr p)
+load_(EVE_SUPPORTS(sve_), C const& cond, safe_type const&, eve::as<wide<T, N>> const&, Ptr p)
 requires(sve_abi<abi_t<T, N>>)
 {
   auto ptr = unalign(p);
 
-  if constexpr( N() == expected_cardinal_v<T> ) return svld1(sve_true<T>(), ptr);
-  else return svldff1(sve_true<T>(), ptr);
+  if constexpr(C::is_complete && C::is_inverted)
+  {
+    if constexpr( N() == expected_cardinal_v<T> ) return svld1(sve_true<T>(), ptr);
+    else return svldff1(sve_true<T>(), ptr);
+  }
+  else
+  {
+    wide<T, N> loaded;
+
+    if constexpr(!C::is_complete)
+    {
+      auto mask = cond.mask(as<wide<T,N>>{});
+      if constexpr( N() == expected_cardinal_v<T> ) loaded = svld1(mask, ptr);
+      else                                          loaded = svldff1(mask, ptr);
+      if constexpr( C::has_alternative )
+      {
+        loaded = svsel(mask, loaded, wide<T,N>(cond.alternative));
+      }
+    }
+    else
+    {
+      if constexpr( C::has_alternative ) loaded = wide<T,N>(cond.alternative);
+    }
+
+    return loaded;
+  }
 }
 
 //================================================================================================
 // Logical support
 //================================================================================================
-template<typename T, typename N, data_source Pointer>
+template<relative_conditional_expr C, typename T, typename N, data_source Pointer>
 EVE_FORCEINLINE logical<wide<T, N>>
-load_ ( EVE_SUPPORTS(sve_), ignore_none_ const&, safe_type const&
-      , eve::as<logical<wide<T, N>>> const&, Pointer ptr
-      ) noexcept
+                load_(EVE_SUPPORTS(sve_),
+                      C const& cond,
+                      safe_type const&,
+                      eve::as<logical<wide<T, N>>> const&,
+                      Pointer ptr) noexcept
 requires(dereference_as<logical<T>, Pointer>::value && sve_abi<abi_t<T, N>>)
 {
-  auto block = load(ignore_none, safe, eve::as<wide<T, N>>{}, ptr_cast<T const>(ptr));
+  auto const alt = [&]()
+  {
+    if constexpr( C::has_alternative ) return cond.rebase(cond.alternative.mask());
```

Comment:
Doesn't rebasr work for no alternative?

Reply:
rebase if the member of the `.else_` subobject so nah.
This should be cleaned up later I guess as this API shoudl eb callable without needing to know about the presence of an alternative.

When called on non-alt conditional, it should just append a .else_()

Reply:
Gonna make a story for that